.circleci/continue_config.yml

```diff
@@ -395,66 +128,6 @@ jobs:
           name: Maybe merge an automatic PR into develop
           command: ./tools/circleci/github_scripts/attempt_merge_pr.sh "merge-main-to-develop" "develop" "${GITHUB_TAGGING_TOKEN}"
 
-  build_docker_image:
-    parameters:
-      python_version:
-        type: string
```

Comment:
After speaking with @antonymilne, the `build_docker_image` job in the `nightly_build_main` workflow was initially designed to pre-build Docker images with all the necessary requirements installed. This was meant to act as a cache to speed up CircleCI tests by avoiding the need to install all dependencies from scratch for every test run. So it should be fine to remove as we do our caching in our nightly builds on GitHub actions.